server/ngwmn/tests/services/test_ngwmn.py

```diff
@@ -6,10 +6,169 @@
 
 import requests as r
 import requests_mock
+import copy
+import ngwmn.services.ngwmn as ngwmn
 
 from ngwmn.services import ServiceException
 from ngwmn.services.ngwmn import (
-    generate_bounding_box_values, get_iddata, get_water_quality, get_well_log)
+    generate_bounding_box_values, get_iddata, get_water_quality, get_well_log, get_statistic, get_statistics)
+from .mock_data import MOCK_WELL_LOG_RESPONSE, MOCK_WQ_RESPONSE, MOCK_OVERALL_STATS, MOCK_MONTHLY_STATS
+
+
+class TestGetStatistics(TestCase):
+
+    def tearDown(self):
+        """
+            This Tear Down replaces ngwmn.get_statistics with the original.
+            Note, that his could be replaced with a with-clause mock.
+        """
+        ngwmn.get_statistic = self.save_ngwmn_get_statistic
+
+    def setUp(self):
+        # save the method for replacement in tearDown
+        self.save_ngwmn_get_statistic = ngwmn.get_statistic
+
+        self.test_service_root = 'http://test.gov'
+        self.test_agency_cd = 'TEST'
+        self.test_site_no = 'TS-42'
+        self.test_site_no_not_ranked = '-'.join([self.test_site_no, 'NOT_RANKED'])
+        self.test_site_no_below = '-'.join([self.test_site_no, 'BELOW'])
+        self.test_site_no_above = '-'.join([self.test_site_no, 'ABOVE'])
+
+        self.overall_not_ranked = copy.copy(MOCK_OVERALL_STATS)
+        self.overall_not_ranked['IS_RANKED'] = 'N'
+        self.overall_ranked_below = copy.copy(MOCK_OVERALL_STATS)
+        self.overall_ranked_below['IS_RANKED'] = 'Y'
+        self.overall_ranked_above = copy.copy(MOCK_OVERALL_STATS)
+        self.overall_ranked_above['IS_RANKED'] = 'Y'
+        self.overall_ranked_above['MEDIATION'] = 'AboveDatum'
+
+        self.test_stats = {self.test_agency_cd: {
+            self.test_site_no_not_ranked: {
+                'wl-overall': self.overall_not_ranked,
+                'site-info': {
+                    'IS_FETCHED': 'Y',
+                    'altDatumCd': 'NGW1701A'
+                },
+                'wl-monthly': {
+                    'N/A': 'not called'
+                }
+            },
+            self.test_site_no_below: {
+                'wl-overall': self.overall_ranked_below,
+                'site-info': {
+                    'IS_FETCHED': 'Y',
+                    'altDatumCd': 'NGW1701B'
+                },
+                'wl-monthly': {
+                    'IS_FETCHED': 'Y',
+                    '1': MOCK_MONTHLY_STATS['1']
+                }
+            },
+            self.test_site_no_above: {
+                'wl-overall': self.overall_ranked_above,
+                'site-info': {
+                    'IS_FETCHED': 'Y',
+                    'altDatumCd': 'NGW1701C'
+                },
+                'wl-monthly': {
+                    'IS_FETCHED': 'Y',
+                    '1': MOCK_MONTHLY_STATS['1']
+                }
+            }
+        }}
+
+    @mock.patch('ngwmn.services.ngwmn.r.get')
```

Comment:
These mocks look good, but I was wondering if there was a reason you choose to mock the import like this rather than using `requests_mock`?

Reply:
First, I did not want to mack the request in the get_statistics because that is calling get_statistic.
I already tested the get_statistic with the request mocking. Within the get_statistics, I want to test it in isolation.

The reason I did it the way I did is because I could not figure out how to use mocking tools because ngwmn.py is not a class structure for easy mocking.